### DISCUSSION TITLE
Age of procedures/sessions

### DISCUSSION THREAD

**Discussion by saskiad** [AllenNeuralDynamics/aind-data-schema#259] [Ideas]
I think a use case will be identifying subjects that were imaged/recorded at specific ages (or age ranges), or had procedures performed at specific ages. We track the dates of these events, and we have the date of birth of the subject. It would be nice to auto-generate the age of the subject in the procedures/session/acquisition files using those values. This requires that the subject dob be accessible - which lives in the subject json. Is there a way to make the different schemas interact in this way?

**Comment by dyf** [comment]
Derived values like this could be deferred to the database layer. e.g. we put special queries into the `aind-data-access-api` to compute these values. The downside to this is that they are not visible to people who consume the JSON directly.

Because the subject, procedure, and session metadata are all generated in different places at different times, creating a hard dependency in order to generate these ages will be finicky to get right. The only reliable place for it to go is the `aind-data-transfer-service`. 

If we ever want to apply code ocean tags based on age at upload time, they would have to be computed before upload.

My instinct is to defer this to query time, but I could be persuaded otherwise.
